src/libjasper/jpc/jpc_dec.c

```diff
@@ -1751,14 +1755,18 @@ static int calcstepsizes(uint_fast16_t refstepsize, int numrlvls,
 {
 	int bandno;
 	int numbands;
+	int r;
+	int nb;
 	uint_fast16_t expn;
 	uint_fast16_t mant;
 	expn = JPC_QCX_GETEXPN(refstepsize);
 	mant = JPC_QCX_GETMANT(refstepsize);
 	numbands = 3 * numrlvls - 2;
 	for (bandno = 0; bandno < numbands; ++bandno) {
 //jas_eprintf("DEBUG %d %d %d %d %d\n", bandno, expn, numrlvls, bandno, ((numrlvls - 1) - (numrlvls - 1 - ((bandno > 0) ? ((bandno + 2) / 3) : (0)))));
-		uint_fast16_t e = expn + (bandno + 2) / 3;
+		r = (bandno > 0) ? ((bandno + 2) / 3) : (0);
```

Comment:
In case of `bandno` equal to 0 the calculation will result in 0:
```suggestion
		r = (bandno + 2) / 3;
```

Reply:
Accepted. Thank you for the suggestion.